Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h

```diff
@@ -97,7 +105,7 @@ class TrackerGeometry final : public TrackingGeometry {
 
   GeomDetEnumerators::SubDetector theSubDetTypeMap[6];
   unsigned int theNumberOfLayers[6];
-
+  std::vector< std::tuple< DetId, TrackerGeometry::ModuleType, float> > theDetTypetList; 
```

Comment:
This vector contains information which are already in the geometry (XML or DB) and it is created once just to make the access to the information easier and faster. I agree with Christophe that there is no need to duplicate the information in the database.

        Andrea

On Jan 28, 2016, at 10:07 , Christophe Delaere <notifications@github.com<mailto:notifications@github.com>>
 wrote:


In Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h<https://github.com/cms-sw/cmssw/pull/12987#discussion_r51095932>:

> @@ -97,7 +105,7 @@ class TrackerGeometry final : public TrackingGeometry {
>
>    GeomDetEnumerators::SubDetector theSubDetTypeMap[6];
>    unsigned int theNumberOfLayers[6];
> -
> +  std::vector< std::tuple< DetId, TrackerGeometry::ModuleType, float> > theDetTypetList;


I think that the use of the db at this stage would be an overkill. This vector is filled once at the beginning from the geometry (which is presently in xml only).

—
Reply to this email directly or view it on GitHub<https://github.com/cms-sw/cmssw/pull/12987/files#r51095932>.